### PR TITLE
test: Port remaining query_tests/src/sql.rs tests to end-to-end

### PR DESCRIPTION
**Pull request by carols10cents**
Connects to #6617.

Despite having ported all the SQL tests in query_tests/cases, there were a few tests remaining in the `query_tests/src/sql.rs` file. These fell into two categories:

- Tests that look like they were written before the `.sql` file test framework existed. There wasn't really a reason these couldn't be written as `.sql` file tests, so I converted them.
- Tests asserting that some SQL resulted in a particular error message. I ported these to `influxdb_iox/tests/query_tests2/sql_errors.rs`.

I also found a bug in the error messages created by the `.sql` test framework code-- if you tried to run a test with a `.sql` file but without any `.expected` file, the error message would say it couldn't find the `.sql` file. That's fixed in the first commit in this PR.

### CODE REVIEW

File: influxdb_iox/tests/query_tests2/sql_errors.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
+//! Tests of SQL queries that are expected to return particular errors.
+
+use crate::query_tests2::{framework::IoxArchitecture, setups::SETUPS};
+use observability_deps::tracing::*;
+use test_helpers_end_to_end::{maybe_skip_integration, MiniCluster, Step, StepTest};
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/sql_errors.rs:7]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L91-L99

File: influxdb_iox/tests/query_tests2/cases/in/restaurant.sql [hunk 2]
```diff
@@ -0,0 +1,7 @@
+-- IOX_SETUP: TwoMeasurementsUnsignedType
+
+-- IOX_COMPARE: sorted
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/restaurant.sql:3]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L102-L117

File: influxdb_iox/tests/query_tests2/cases/in/restaurant.sql [hunk 3]
```diff
@@ -0,0 +1,7 @@
...
+SELECT town, count from restaurant;
+
+-- IOX_COMPARE: sorted
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/restaurant.sql:6]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L120-L135

File: influxdb_iox/tests/query_tests2/cases/in/union_all.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+-- IOX_SETUP: TwoMeasurementsManyFields
+
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/union_all.sql:2]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L138-L160

File: influxdb_iox/tests/query_tests2/cases/in/duplicates_ingester.sql [hunk 5]
```diff
@@ -1,11 +1,13 @@
 -- Test for predicate push down explains
 -- IOX_SETUP: OneMeasurementFourChunksWithDuplicatesWithIngester
 
+-- IOX_COMPARE: sorted
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/duplicates_ingester.sql:4]: This was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L223-L251 and it looked like the same query being used with EXPLAIN here, so I added it without EXPLAIN in both this `duplicates_ingester.sql` file and the `duplicates_parquet.sql` file

File: influxdb_iox/tests/query_tests2/cases/in/aggregates_with_nulls.sql [hunk 6]
```diff
@@ -0,0 +1,9 @@
+-- IOX_SETUP: TwoMeasurementsManyNulls
+
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/aggregates_with_nulls.sql:2]: Was  https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L163-L179

File: influxdb_iox/tests/query_tests2/cases/in/schema_merge.sql [hunk 7]
```diff
@@ -0,0 +1,8 @@
+-- IOX_SETUP: MultiChunkSchemaMerge
+
+-- IOX_COMPARE: sorted
+SELECT * from cpu;
+
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/schema_merge.sql:5]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L203-L220

File: influxdb_iox/tests/query_tests2/cases/in/schema_merge.sql [hunk 8]
```diff
@@ -0,0 +1,8 @@
+-- IOX_SETUP: MultiChunkSchemaMerge
+
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/schema_merge.sql:2]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L76-L88

File: influxdb_iox/tests/query_tests2/cases/in/aggregates_with_nulls.sql [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+-- Validate distinct aggregates work against dictionary columns which have nulls in them
+-- IOX_COMPARE: sorted
+SELECT count(DISTINCT city) FROM o2;
+
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/aggregates_with_nulls.sql:6]: Was  https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L182-L200

File: influxdb_iox/tests/query_tests2/cases/in/different_tag_sets.sql [hunk 10]
```diff
@@ -0,0 +1,7 @@
+-- IOX_SETUP: OneMeasurementTwoChunksDifferentTagSet
+
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/different_tag_sets.sql:2]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L254-L265

File: influxdb_iox/tests/query_tests2/cases/in/different_tag_sets.sql [hunk 11]
```diff
@@ -0,0 +1,7 @@
...
+-- IOX_COMPARE: sorted
+SELECT temp FROM h2o;
+
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/different_tag_sets.sql:5]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L268-L285

File: influxdb_iox/tests/query_tests2/setups.rs [hunk 12]
```diff
@@ -366,6 +366,18 @@ pub static SETUPS: Lazy<HashMap<SetupName, SetupSteps>> = Lazy::new(|| {
                 },
             ],
         ),
+        (
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/setups.rs:369]: So this setup actually corresponds to the setup formerly named `NoDeleteOneChunk` https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/scenarios/delete.rs#L65-L82 but given that the RPC write architecture doesn't support delete predicates yet, and it was being used here without anything to do with deletes, I renamed it.

File: influxdb_iox/tests/query_tests2/cases/in/aggregates.sql [hunk 13]
```diff
@@ -0,0 +1,10 @@
+-- IOX_SETUP: OneMeasurementWithTags
+
+-- IOX_COMPARE: sorted
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/aggregates.sql:3]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L290-L305

File: influxdb_iox/tests/query_tests2/cases/in/aggregates.sql [hunk 14]
```diff
@@ -0,0 +1,10 @@
+-- IOX_SETUP: OneMeasurementWithTags
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/aggregates.sql:1]: See note in setups.rs about why this name doesn't match up

File: influxdb_iox/tests/query_tests2/cases/in/aggregates.sql [hunk 15]
```diff
@@ -0,0 +1,10 @@
...
+
+-- IOX_COMPARE: sorted
+SELECT count(time), count(*), count(bar), min(bar), max(bar), min(time), max(time) FROM cpu;
+
+-- IOX_COMPARE: sorted
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/aggregates.sql:6]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L308-L322

File: influxdb_iox/tests/query_tests2/cases/in/aggregates.sql [hunk 16]
```diff
@@ -0,0 +1,10 @@
...
+SELECT max(foo) FROM cpu;
+
+-- IOX_COMPARE: sorted
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/cases/in/aggregates.sql:9]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L325-L339

File: influxdb_iox/tests/query_tests2/sql_errors.rs [hunk 17]
```diff
@@ -0,0 +1,139 @@
...
+        sql: "SELECT * from cpu where foo = 8",
+        expected_error_code: tonic::Code::InvalidArgument,
+        expected_message: "Error while planning query: Schema error: No field named 'foo'. \
+        Valid fields are 'cpu'.'host', 'cpu'.'region', 'cpu'.'system', 'cpu'.'time', 'cpu'.'user'.",
+    }
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/sql_errors.rs:20]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L342-L351

File: influxdb_iox/tests/query_tests2/sql_errors.rs [hunk 18]
```diff
@@ -0,0 +1,139 @@
...
+        expected_error_code: tonic::Code::Internal,
+        expected_message: "Error while planning query: Internal error: Unsupported logical plan: \
+        CreateExternalTable. This was likely caused by a bug in DataFusion's code and we would \
+        welcome that you file an bug report in our issue tracker",
+    }
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/sql_errors.rs:38]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L354-L363

File: influxdb_iox/tests/query_tests2/sql_errors.rs [hunk 19]
```diff
@@ -0,0 +1,139 @@
...
+        sql: "CREATE SCHEMA foo",
+        expected_error_code: tonic::Code::InvalidArgument,
+        expected_message: "Error while planning query: This feature is not implemented: \
+        CreateCatalogSchema",
+    }
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/sql_errors.rs:55]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L366-L374

File: influxdb_iox/tests/query_tests2/sql_errors.rs [hunk 20]
```diff
@@ -0,0 +1,139 @@
...
+        sql: "SELECT selector_last(time)['bar'] FROM cpu",
+        expected_error_code: tonic::Code::InvalidArgument,
+        expected_message: "Error while planning query: Error during planning: selector_last \
+        requires exactly 2 arguments, got 1",
+    }
+    .run()
+    .await;
+}
+
+#[tokio::test]
```
Comment (carols10cents) [inline influxdb_iox/tests/query_tests2/sql_errors.rs:68]: Was https://github.com/influxdata/influxdb_iox/blob/0ef4f9fa9b770aa646f75c6670f1ce2b7892ada1/query_tests/src/sql.rs#L377-L385

**Review by carols10cents — COMMENTED** [review]
Matching up tests

**Review by alamb — APPROVED** [review]
This looks great -- thank you @carols10cents 

One more to go 🤞

**Pull request merged**